### ISSUE TITLE
Issues with ImageReader

### ISSUE THREAD

**Issue by yefeng-zheng** [Microsoft/CNTK#263]
I am experimenting with ImageReader and found a few issues.
1. ImageReader doesn't work with gray-scale images with only one channel.
   I adapted both the MNIST and CIFAR networks, making necessary changes. CNTK doesn't crash. The training error is good. However, the test error is very bad. If I convert the data to 3-channel images (still gray-scale since the R, G, B channels are duplicated), I can train a good network with the same training parameters.
2. Training with ImageReader on Windows is extremely slow
   I tried example \Examples\Image\Miscellaneous\CIFAR-10\03_ResNet.cntk. Training on Windows is 10x slower than Linux. So, basically, ImageReader is useless on Windows.

BTW - There are many readers available in the source code under \Source\Readers. However, many of them have no documentation at all (e.g., BinaryReader and LibSVMBinaryReader). Therefore, I don't know how to use them. Would you please add more documentation or provide a converter? Thank you very much!

**Comment by wolfma61** [comment]
we are currently doing major work on readers (some information is/will show up in the wiki very shortly). There will be Image, binary, text, speech readers (you see some ongoing work in branches) as well as a more generic text format to use

this should materialize in the very near future

**Comment by Alexey-Kamenev** [comment]
Hello,
ImageReader being 10x slower on Windows is certainly not expected. Can you please share the numbers you see in the log (search for `SamplesPerSecond`). For example, on my Win machine with TitanX it is around 4100 for CIFAR-10 03_ResNet config (which is a 20-layer ResNet) and around 750 for ResNet-56. On Linux I see about the same numbers.

**Comment by yefeng-zheng** [comment]
Previously, I tested "Examples\Image\Miscellaneous\CIFAR-10\03_ResNet.cntk" on Windows 7. Today, I tested again on Windows Server 2012. The training speed on Windows Server 2012 is also much slower than Linux. Following is the speed of the first epoch, taking 215 seconds. On a Linux computer, it takes only around 20 seconds. I didn't make any changes. So, it is very easy for you to verify. I have an NVIDIA Titan X GPU and it is rarely used (~10% usage). Probably, it is the reason why the training is so slow.


Starting Epoch 1: learning rate per sample = 0.007813  effective momentum = 0.900000  momentum as time constant = 1214.9 samples

Starting minibatch loop.
 Epoch[ 1 of 160]-Minibatch[   1- 200]: SamplesSeen = 25600; TrainLossPerSample =  1.79306732; EvalErr[0]PerSample = 0.67878906; TotalTime = 117.5910s; SamplesPerSecond = 217.7
Finished Epoch[ 1 of 160]: [Training Set] TrainLossPerSample = 1.5940382; TotalSamplesSeen = 50000; EvalErrPerSample = 0.59329998; AvgLearningRatePerSample = 0.0078125; EpochTime=214.908
SGD: Saving checkpoint model './Output/Models/03_ResNet.1'

**Comment by Alexey-Kamenev** [comment]
This is strange indeed. This is what I get on my WinServer 2012 R2 with Titan X:
```
Starting minibatch loop.
 Epoch[14 of 160]-Minibatch[   1- 200, 0.0000000000006%]: SamplesSeen = 25600; TrainLossPerSample =  0.43106258; EvalErr[0]PerSample = 0.14972656; TotalTime = 5.6830s; SamplesPerSecond = 4504.7
Finished Epoch[14 of 160]: [Training Set] TrainLossPerSample = 0.43401468; TotalSamplesSeen = 700000; EvalErrPerSample = 0.15116; AvgLearningRatePerSample = 0.0078125; EpochTime=11.1002
SGD: Saving checkpoint model './Output/Models/03_ResNet.14'
```
So in your case `SamplesPerSecond = 217.7` while in my: `SamplesPerSecond = 4504.7`. What happens if you let it run for a few epochs?

**Comment by pkranen** [comment]
The readers have been reworked. Please try with the latest bits open a new issue if required.
